db/imgcodec/tiff/bitdepths/make.py

```diff
@@ -17,7 +18,23 @@ def filename(bits):
 def ref_filename(bits):
     return os.path.join(ref_path, f'rgb_{bits}bit.tiff.npy')
 
-for bits in (8, 16, 32):
-    uint_image = (image * (2**bits - 1)).astype(f'uint{bits}')
-    tifffile.imwrite(filename(bits), uint_image)
-    np.save(ref_filename(bits), uint_image)
+def ref_float_filename(bits):
+    return os.path.join(ref_path, f'rgb_{bits}bit_float.tiff.npy')
+
+standard_bitdepths = [8, 16, 32]
+
+for bits in range(1, 32+1):
+    ref = np.round(image * (2**bits - 1)) / (2**bits - 1)
+    np.save(ref_float_filename(bits), ref)
```

Comment:
Is this enough to store the reference as float? Or maybe we should keep `(image * (2**bits - 1)).astype(f'uint32')`?

Reply:
The bitdepth conversion is not exact in DALI (we cast to double, multiply by a coefficient, and then round back to an integer), so in the tests we check for being _close_ with some epsilon, so I don't see a need to have the reference as exact `uint32`.